pems-server/src/main/java/org/maroubra/pemsserver/monitoring/utsapi/WebSensor.java

```diff
@@ -0,0 +1,80 @@
+package org.maroubra.pemsserver.monitoring.utsapi;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class WebSensor {
+    private LocalDateTime fromDate;
+    private LocalDateTime toDate;
+    private int pollIntervalMinutes = 60;
+    private String family;
+    private String subSensor;
+    private String sensor;
+    private String apiLink = "http://eif-research.feit.uts.edu.au/api/";
+    private static Logger log = LoggerFactory.getLogger(WebSensor.class);
+
+    private Retrofit retrofit = new Retrofit.Builder()
+            .baseUrl(apiLink)
+            .addConverterFactory(GsonConverterFactory.create())
+            .build();
+
+    private WebApiRequest service = retrofit.create(WebApiRequest.class);
```

Comment:
For inversion of control (IoC) and unit testing purposes, it would be good for retrofit and the api to be created outside the class and injected through the constructor as a dependency.